src/js/stores/BallotStore.js

```diff
@@ -127,13 +133,26 @@ class BallotStore extends FluxMapStore {
       case "voterAddressSave":
         key = action.res.google_civic_election_id;
         newBallot[key] = action.res;
+        // console.log("BallotStore, voterAddressSave, key: " + key);
 
+        // We need to refresh the ballot here after the address is changed. If we don't, it is possible to change
+        // the address and be left with the ballot for the prior address.
+        BallotActions.retrieve();
```

Comment:
I think we need to talk through the dependencies in BallotStore to reach the real solution to this issue, and that requires that we have a conversation. Unfortunately given the deadline and all of our schedules, at the earliest we will be able to have that conversation Saturday morning. I think it is very risky to leave this highly visible bug in place until Saturday morning given we will be showing the app to a number of new people early Saturday morning at NDoCH. I think we should work with the solution in this pull request, and then treat the next step as optimization, refactoring so that we don't need to make a second or third call in certain circumstances. I am not a fan of the extra API calls included in this pull request, but what I have checked in works in all of the situations I tested. Removing any of these extra "BallotActions.retrieve();" lines causes the issue to be apparent to a voter in at least one of the cases I tested.